src/spsa/README.md

```diff
@@ -0,0 +1,240 @@
+# SPSA tuning using OpenBench
+
+This README will not explain how the SPSA tuning works, or why we use it. 
+For that please read the WIKI from [OpenBench](https://github.com/AndyGrant/OpenBench/wiki/SPSA-Tuning-Workloads).
+
+## How to use
+
+### First step
+
+Modify the `Makefile` on the test branch to include spsa option:
+
+```diff
+openbench:
++	meson setup .openbench --cross-file targets/linux-native.txt --wipe --buildtype=release -Dspsa=true
+-	meson setup .openbench --cross-file targets/linux-native.txt --wipe --buildtype=release
+	meson compile -C .openbench
+	cp .openbench/meltdown-chess-engine $(EXE)
+```
+
+It's fine to commit this to your local SPSA branch as it will only be used for SPSA testing anyways.
+I'm hoping to add the compilation flag to OpenBench at one point, but for now this should do.
+
+### Second step: 
+
+Compile meltdown locally with SPSA enabled: 
+```
+./scripts/build.sh --spsa -r
+```
+(`--spsa` will enable SPSA and `-r` will run the executable when built).
+
+### Third step 
+
+Now run `spsa` from meltdown itself:
+```
+spsa
+fullDepthMove, int, 4.0, 0.0, 12.0, 1.0, 0.002
+lmrReductionLimit, int, 3.0, 0.0, 12.0, 1.0, 0.002
+rfpReductionLimit, int, 3.0, 0.0, 12.0, 1.0, 0.002
+rfpMargin, int, 100.0, 0.0, 150.0, 10.0, 0.002
+rfpEvaluationMargin, int, 120.0, 0.0, 150.0, 10.0, 0.002
+razorReductionLimit, int, 3.0, 0.0, 12.0, 1.0, 0.002
+razorMarginShallow, int, 125.0, 0.0, 250.0, 10.0, 0.002
+razorMarginDeep, int, 175.0, 0.0, 250.0, 10.0, 0.002
+razorDeepReductionLimit, int, 2.0, 0.0, 12.0, 1.0, 0.002
+nullMoveReduction, int, 2.0, 0.0, 12.0, 1.0, 0.002
+aspirationWindow, int, 50.0, 10.0, 100.0, 5.0, 0.002
+timeManMovesToGo, int, 20.0, 10.0, 40.0, 2.0, 0.002
+timeManHardLimit, int, 3.0, 1.0, 5.0, 1.0, 0.002
+seePawnValue, int, 100.0, 50.0, 1000.0, 10.0, 0.002
+seeKnightValue, int, 422.0, 50.0, 1000.0, 10.0, 0.002
+seeBishopValue, int, 422.0, 50.0, 1000.0, 10.0, 0.002
+seeRookValue, int, 642.0, 50.0, 1000.0, 10.0, 0.002
+seeQueenValue, int, 1015.0, 50.0, 1500.0, 10.0, 0.002
+```
+The output will be the expected SPSA input for OpenBench.
+
+### Fourth step
+
+Start the tuning from OpenBench:
+
+Pick `Create Tune` in the side menu.
+
+Pick `Meltdown` in engine list.
+
+Setup tuning (default should be a fine starting point).
+
+Paste the SPSA input:
+![image](https://github.com/user-attachments/assets/b945bc7d-db42-4e77-815f-507de9129ec7)
+
+### Fifth step
+
+Start the tuner and wait :)
+
+The tuning will take a few days so please be patient!
+
+## SPSA parameters
+
+The content in `parameters.h` is grouped to have a single point of entry where we can add tuning parameters.
+When a parameter has been added then it will be reflected in a few different places.

```

Comment:
I know we're not explaining the motivation for SPSA, but do you think it's worth briefly what makes a new parameter a candidate for SPSA tuning?

Reply:
Tbh I'm a bit unsure on that part myself haha..
I think some trial and error is gonna do wonder here :)

Reply:
Probably also doesn't make sense to tune all parameters at once every time. 

Might make sense to do an isolated SEE value tuning, isolated time manager tuning, and the leave the magic numbers in the search/pruning for themselves. But, still don't know 😆

Reply:
I updated the readme - a bit more clarification :+1: